src/trie.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+from collections import deque, OrderedDict
+
+
+class Trie(object):
+    def __init__(self, iterable=None):
+        self.root = OrderedDict()
+        if iterable:
+            for i in iterable:
+                self.insert(i)
+
+    def insert(self, token):
+        start = self.root
+        for letter in token:
+            start = start.setdefault(letter, OrderedDict())
+        start['#'] = '#'
+
+    def contains(self, token):
+        """Verify if a token is in the trie."""
+        start = self.root
+        counter = 0
+        while counter < len(token):
+            try:
+                start = start[token[counter]]
+            except KeyError:
+                return False
+            counter += 1
+            try:
+                return start['#'] == '#'
```

Comment:
This seems out of place here. It would be more explicit and assertive to do this check only once you've reached the very end of your token, so you check for sure once you exit the loop that the end-of-word char is sitting there.

Otherwise it seems you're taking an approach of "I know this is gonna work so I'll just leave it in there."

This will fail in a certain case. Suppose I insert the word `"Python"` into the tree. When you call `contains("Pythonista")`, it will return `True` though I have not inserted that word.

On that note, also test for a scenario like the above to make sure `contains("Py")` returns `False`.